core/shared/src/main/scala/org/typelevel/idna4s/core/uts46/CodePointMapper.scala

```diff
@@ -0,0 +1,220 @@
+            } else {
+              loop(acc.appendCodePoint(value), errors, index + indexIncrement)
+            }
+          case -7 => // DEVIATION_MULTI
+            if (transitionalProcessing) {
+              loop(unsafeLookupDeviationCodePointMultiMapping0(value).foldLeft(acc){
+                case (acc, value) =>
+                  acc.appendCodePoint(value)
+              }, errors, index + indexIncrement)
+            } else {
+              loop(acc.appendCodePoint(value), errors, index + indexIncrement)
+            }
+          case -8 => // DISALLOWED
+            loop(acc.appendCodePoint(ReplacementCharacter), MappingError(index, "Disallowed code point in input.", CodePoint.unsafeFromInt(value)) :: errors, index + indexIncrement)
+          case -9 => // DISALLOWED_STD3_VALID
+            if (useStd3ASCIIRules) {
+              loop(acc.appendCodePoint(ReplacementCharacter), MappingError(index, "Disallowed code point in input.", CodePoint.unsafeFromInt(value)) :: errors, index + indexIncrement)
+            } else {
+              loop(acc.appendCodePoint(value), errors, index + indexIncrement)
+            }
+          case -10 => // DISALLOWED_STD3_MAPPED
+            if (useStd3ASCIIRules) {
+              loop(acc.appendCodePoint(ReplacementCharacter), MappingError(index, "Disallowed code point in input.", CodePoint.unsafeFromInt(value)) :: errors, index + indexIncrement)
+            } else {
+              loop(acc.appendCodePoint(unsafeLookupDisallowedSTD3CodePointMapping0(value)), errors, index + indexIncrement)
+              }
+          case -11 => // DISALLOWED_STD3_MAPPED_MULTI
+            if (useStd3ASCIIRules) {
+              loop(acc.appendCodePoint(ReplacementCharacter), MappingError(index, "Disallowed code point in input.", CodePoint.unsafeFromInt(value)) :: errors, index + indexIncrement)
+            } else {
+              loop(unsafeLookupDisallowedSTD3CodePointMultiMapping0(value).foldLeft(acc){
+                case (acc, value) =>
+                  acc.appendCodePoint(value)
+              }, errors, index + indexIncrement)
+            }
+          case -12 => // DEVIATION_IGNORED
+            loop(acc, errors, index + indexIncrement)
+          case otherwise => // MAPPED or bug
+            if (otherwise < 0) {
+              throw new AssertionError(s"Unexpected mapping result (this is probably a bug in idna4s): $otherwise")
+            } else {
+              // This is a mapped code point
+              loop(acc.appendCodePoint(otherwise), errors, index + indexIncrement)
+            }
+        }
+      }
+
+    loop(new StringBuilder(len), Nil, 0)
+  }
+
+  def mapCodePoint(codePoint: CodePoint): CodePointStatus = {
+    import CodePointStatus._
+
+    (unsafeMapIntCodePointSentinel0(codePoint.value): @switch) match {
+      // Literals used rather than named constants because scalac won't
+      // generate a switch with the named constants.
+      case -1 => // VALID
```

Comment:
Huh, that's annoying. I wonder if the `final val` in the `object` would make a difference, but I'm doubtful.

Reply:
I'll try `final val` and check.